libs/pika/ini/src/ini.cpp

```diff
@@ -210,7 +210,8 @@ namespace pika::detail {
                 for (std::string::size_type pos1 = sec_name.find_first_of('.');
                      std::string::npos != pos1; pos1 = sec_name.find_first_of('.', pos = pos1 + 1))
                 {
-                    current = current->add_section_if_new(sec_name.substr(pos, pos1 - pos));
+                    current = current->add_section_if_new(
+                            sec_name.substr(pos, static_cast<std::string::difference_type>(pos1 - pos)));
```

Comment:
How does this fix

> Unsanitized input from a file flows into a subtraction operator (-), where it is used in integer arithmetic. This may result in an integer overflow vulnerability.

?

Reply:
Is snyk any happier with the current version? It's at least a lot more readable to me!